lib/avo/resources/resource_manager.rb

```diff
@@ -35,37 +34,30 @@ def build
         #   "FishResource",
         # ]
         def fetch_resources
-          resources = if Avo.configuration.resources.nil?
-            BaseResource.descendants
+          if Avo.configuration.resources.present?
+            load_configured_resources
           else
-            Avo.configuration.resources
+            load_resources_namespace
           end
 
-          resources.map do |resource|
-            if resource.is_a?(Class)
-              resource
-            else
-              resource.to_s.safe_constantize
-            end
+          BaseResource.descendants
+        end
+
+        def load_resources_namespace
+          Rails.autoloaders.main.eager_load_namespace(Avo::Resources)
```

Comment:
`eager_load_namespace` is idempotent, safe to re-run, and short circuits if full eager loading is detected.